includes/admin/discounts/class-discount-codes-table.php

```diff
@@ -189,55 +203,111 @@ protected function get_primary_column_name() {
 	 * @access public
 	 * @since 1.4
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param EDD_Discount $item Discount object.
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
 	 */
 	function column_default( $item, $column_name ) {
-		return $item[ $column_name ];
+		return property_exists( $item, $column_name ) ? $item->$column_name : '';
+	}
+
+	/**
+	 * This function renders the amount column.
+	 *
+	 * @access public
+	 * @since 3.0
+	 *
+	 * @param EDD_Discount $item Data for the discount code.
+	 * @return string Formatted amount.
+	 */
+	public function column_amount( $item ) {
+		return edd_format_discount_rate( $item->type, $item->amount );
+    }
+
+	/**
+	 * This function renders the start column.
+	 *
+	 * @access public
+	 * @since 3.0
+	 *
+	 * @param EDD_Discount $item Discount object.
+	 * @return string Start  date
+	 */
+	function column_start_date( $item ) {
```

Comment:
We should add access modifiers to this, `column_default`, `column_end_date`, `column_name`, `column_cb`, `column_status`, and `no_items`. PhpStorm is whining that they're not declared `public`, `protected`, whatever.

Reply:
Access modifiers have been added.